device/Microsoft.Azure.Devices.Client.Shared/IoTHubClientDiagnostic.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace Microsoft.Azure.Devices.Client
+{
+    using System;
+    using System.Globalization;
+    using Microsoft.Azure.Amqp;
+
+    class IoTHubClientDiagnostic
+    {
+        const string Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+        const string DiagnosticCreationTimeUtcKey = "creationtimeutc";
+
+        internal const string MqttDiagIdKey = "$.diagid";
```

Comment:
Please split Mqtt / AMQP and avoid placing all items in a single class. Each class should respect the single responsibility pattern.
I would move these within the classes where they are actually used instead of here.
`MqttDiagIdKey`, `MqttDiagCorrelationContextKey`, `AmqpDiagIdKey`, `AmqpDiagCorrelationContextKey`

Reply:
Yeah, I will move these to the class where they are actually used.